paddle/framework/framework.proto

```diff
@@ -143,3 +143,13 @@ message BlockDesc {
 // https://github.com/PaddlePaddle/Paddle/blob/develop/doc/design/program.md
 // for more details.
 message ProgramDesc { repeated BlockDesc blocks = 1; }
+
+message OpPlacement {
+  optional string name = 1;
+  optional string device = 2;
+}
+
+message ExecutionPlan {
+  optional ProgramDesc program = 1;
```

Comment:
Pserver and trainer may use different `ProgramDesc`, seems one field is not enough?

Reply:
In the new design the pserver and trainer will be exactly same binary (executor), only thing that is different is the `ExecutionPlan` they run. The `Planner` will know the roles of different executor (e.g., pserver role, trainer role) to help generating the `ExecutionPlans`.